src/grpc_interceptor/testing/dummy_client.py

```diff
@@ -27,55 +28,68 @@ class DummyService(dummy_pb2_grpc.DummyServiceServicer):
             This allows testing special cases, like raising exceptions.
     """
 
-    def __init__(self, special_cases: Dict[str, SpecialCaseFunction]):
+    def __init__(
+        self,
+        special_cases: Dict[str, SpecialCaseFunction],
+        context_cases: Optional[Dict[str, SpecialCaseFunction]] = None,
```

Comment:
Rather than add a new `context_cases` parameter, maybe we should extend `SpecialCaseFunction` to have type `Callable[[str, grpc.ServicerContext], str]`. It's not backwards compatible, but this package is not that widely used right now and it would only break tests.

Reply:
Yeah that makes sense... Also allows the handler to always have both the input and context to do with as it pleases.

Reply:
Done.